awscli/customizations/s3/s3handler.py

```diff
@@ -385,151 +396,112 @@ def __init__(self, session, params, result_queue=None,
+        :param manager: The transfer manager to use for the upload.
 
-        # Submit download part tasks to the executor.
-        self._do_enqueue_range_download_tasks(
-            filename=filename, chunksize=self.chunksize,
-            num_downloads=num_downloads, context=context,
-            remove_remote_file=remove_remote_file
-        )
-        return num_downloads
+        :type bucket: str
+        :param bucket: The bucket to upload the stream to.
 
-    def _enqueue_multipart_upload_tasks(self, filename, payload=None):
-        # First we need to create a CreateMultipartUpload task,
-        # then create UploadTask objects for each of the parts.
-        # And finally enqueue a CompleteMultipartUploadTask.
+        :type key: str
+        :param key: The name of the key to upload the stream to.
+
+        :return: A CommandResult representing the upload status.
+        """
+        # Upload chunksizes are restricted, so we need to do some massaging
+        # to get them right.
         if self.params['expected_size']:
             # If we have the expected size, we can calculate an appropriate
             # chunksize based on max parts and chunksize limits
-            chunksize = find_chunksize(int(self.params['expected_size']),
-                                       self.chunksize)
+            chunksize = find_chunksize(
+                int(self.params['expected_size']),
+                self.config.multipart_chunksize)
         else:
             # Otherwise, we can still adjust for chunksize limits
-            chunksize = adjust_chunksize_to_upload_limits(self.chunksize)
+            chunksize = adjust_chunksize_to_upload_limits(
+                self.config.multipart_chunksize)
+        self.config.multipart_chunksize = chunksize
+
+        stream_filein = sys.stdin
+        if six.PY3:
+            # This is to get a binary stdin
+            stream_filein = sys.stdin.buffer
 
-        num_uploads = '...'
+        params = {}
+        RequestParamsMapper.map_put_object_params(params, self.params)
 
-        # Submit a task to begin the multipart upload.
-        upload_context = self._enqueue_upload_start_task(
-            chunksize, num_uploads, filename)
+        future = manager.upload(
+            fileobj=stream_filein, bucket=bucket,
+            key=key, extra_args=params)
 
-        # Now submit a task to upload the initial chunk of data pulled
-        # from the stream that was used to determine if a multipart upload
-        # was needed.
-        self._enqueue_upload_single_part_task(
-            part_number=1, chunk_size=chunksize,
-            upload_context=upload_context, filename=filename,
-            task_class=tasks.UploadPartTask, payload=payload
-        )
+        # Set the size of the transfer if expected size is set.
+        expected_size = self.params.get('expected_size', None)
+        if expected_size is not None:
+            future.meta.size = expected_size
```

Comment:
You should not be able to do that...https://github.com/boto/s3transfer/blob/develop/s3transfer/futures.py#L92 You need to use a subscriber and the ``provide_transfer_size`` method in the ``on_queued()`` method.